connexion does not parse simple reference links

### Description

My OpenAPI 3.0.0 specification contains entries that look like this:

```
$ref: 'models/provider.yaml#/components/schemas/Provider'
```
This is "remote reference" as documented in the [OpenAPI documentation](https://swagger.io/docs/specification/using-ref/).

### Expected behaviour

I expect connexion to resolve the link and read the specified content from the specified file.

### Actual behaviour

A positively ungodly traceback:

```
>>> app.add_api('api-marketplace.yaml')
Traceback (most recent call last):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 380, in resolve_from_url
    document = self.store[url]
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_utils.py", line 23, in __getitem__
    return self.store[self.normalize(uri)]
KeyError: 'models/provider.yaml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 383, in resolve_from_url
    document = self.resolve_remote(url)
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 474, in resolve_remote
    result = json.loads(urlopen(uri).read().decode("utf-8"))
  File "/usr/lib64/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.7/urllib/request.py", line 510, in open
    req = Request(fullurl, data)
  File "/usr/lib64/python3.7/urllib/request.py", line 328, in __init__
    self.full_url = url
  File "/usr/lib64/python3.7/urllib/request.py", line 354, in full_url
    self._parse()
  File "/usr/lib64/python3.7/urllib/request.py", line 383, in _parse
    raise ValueError("unknown url type: %r" % self.full_url)
ValueError: unknown url type: 'models/provider.yaml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/apps/flask_app.py", line 54, in add_api
    api = super(FlaskApp, self).add_api(specification, **kwargs)
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/apps/abstract.py", line 155, in add_api
    options=api_options.as_dict())
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/apis/abstract.py", line 73, in __init__
    self.specification = Specification.load(specification, arguments=arguments)
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/spec.py", line 151, in load
    return cls.from_file(spec, arguments=arguments)
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/spec.py", line 108, in from_file
    return cls.from_dict(spec)
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/spec.py", line 146, in from_dict
    return OpenAPISpecification(spec)
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/spec.py", line 39, in __init__
    self._validate_spec(raw_spec)
  File "/home/lars/.local/lib/python3.7/site-packages/connexion/spec.py", line 230, in _validate_spec
    validate_spec(spec)
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/shortcuts.py", line 7, in validate
    return validator_callable(spec, spec_url=spec_url)
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/validators.py", line 47, in validate
    for err in self.iter_errors(spec, spec_url=spec_url):
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 58, in wrapper
    for err in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/validators.py", line 56, in iter_errors
    for err in validator.iter_errors(spec):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 304, in properties_draft4
    schema_path=property,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 16, in patternProperties
    v, subschema, path=k, schema_path=pattern,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 304, in properties_draft4
    schema_path=property,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 304, in properties_draft4
    schema_path=property,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 16, in patternProperties
    v, subschema, path=k, schema_path=pattern,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 341, in oneOf_draft4
    errs = list(validator.descend(instance, subschema, schema_path=index))
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 304, in properties_draft4
    schema_path=property,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 29, in additionalProperties
    for error in validator.descend(instance[extra], aP, path=extra):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 216, in ref
    for error in validator.descend(instance, resolved):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 22, in wrapped
    for res in func(validator, schema_element, instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/_validators.py", line 304, in properties_draft4
    schema_path=property,
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 121, in descend
    for error in self.iter_errors(instance, schema):
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 105, in iter_errors
    for error in errors:
  File "/home/lars/.local/lib/python3.7/site-packages/openapi_spec_validator/decorators.py", line 34, in wrapped
    with self.instance_resolver.resolving(ref) as target:
  File "/usr/lib64/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 366, in resolving
    url, resolved = self.resolve(ref)
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 375, in resolve
    return url, self._remote_cache(url)
  File "/home/lars/.local/lib/python3.7/site-packages/jsonschema/validators.py", line 385, in resolve_from_url
    raise RefResolutionError(exc)
jsonschema.exceptions.RefResolutionError: unknown url type: 'models/provider.yaml'
```

### Additional info:

Output of the commands:

- `python --version`: `Python 3.7.3`
- `pip show connexion | grep "^Version\:"`: `Version: 2.2.0` (although this was in fact installed from  0a05243)

I have the exact same issue when trying to reference a yaml file that is in the same location as the current file.  Is there any solution or workaround to this?

```
             schema:
                type: array
                items:
                  $ref: 'components.yaml#/components/schemas/Warehouse'
```

```
service_1            | 2020-01-22 14:05:38 [11] [DEBUG] connexion.apis.abstract Loading specification: /app/wms_automation_platform/services/specifications/api.yaml
service_1            | 2020-01-22 14:05:38 [12] [DEBUG] openapi_spec_validator.decorators Attaching x-scope to {'$ref': 'components.yaml#/components/schemas/Warehouse'}
service_1            | [2020-01-22 14:05:38 +0000] [12] [ERROR] Exception in worker process
service_1            | Traceback (most recent call last):
service_1            |   File "/pyenv/versions/wms-automation-platform/lib/python3.8/site-packages/jsonschema/validators.py", line 380, in resolve_from_url
service_1            |     document = self.store[url]
service_1            |   File "/pyenv/versions/wms-automation-platform/lib/python3.8/site-packages/jsonschema/_utils.py", line 23, in __getitem__
service_1            |     return self.store[self.normalize(uri)]
service_1            | KeyError: 'components.yaml'
```

If I prefix the URI with file: it gets closer

```
              schema:
                type: array
                items:
                  $ref: 'file:components.yaml#/components/schemas/Warehouse'
```

```
service_1            | 2020-01-22 14:19:52 [11] [DEBUG] connexion.apis.abstract Loading specification: /app/wms_automation_platform/services/specifications/api.yaml
service_1            | 2020-01-22 14:19:53 [10] [DEBUG] openapi_spec_validator.decorators Attaching x-scope to {'$ref': 'file:components.yaml#/components/schemas/Warehouse'}
service_1            | [2020-01-22 14:19:53 +0000] [10] [ERROR] Exception in worker process
service_1            | Traceback (most recent call last):
service_1            |   File "/pyenv/versions/wms-automation-platform/lib/python3.8/site-packages/jsonschema/validators.py", line 380, in resolve_from_url
service_1            |     document = self.store[url]
service_1            |   File "/pyenv/versions/wms-automation-platform/lib/python3.8/site-packages/jsonschema/_utils.py", line 23, in __getitem__
service_1            |     return self.store[self.normalize(uri)]
service_1            | KeyError: 'file:///components.yaml'
service_1            |
service_1            | During handling of the above exception, another exception occurred:
service_1            |
service_1            | Traceback (most recent call last):
service_1            |   File "/pyenv/versions/3.8.0/lib/python3.8/urllib/request.py", line 1474, in open_local_file
service_1            |     stats = os.stat(localfile)
service_1            | FileNotFoundError: [Errno 2] No such file or directory: '/components.yaml'
```

--- a/connexion/json_schema.py
+++ b/connexion/json_schema.py
@@ -62,14 +62,15 @@ class URLHandler:
             return yaml.load(fh, ExtendedSafeLoader)
 
 
-default_handlers = {
+handlers = {
     "http": URLHandler(),
     "https": URLHandler(),
     "file": FileHandler(),
+    "": FileHandler(),
 }
 
 
-def resolve_refs(spec, store=None, handlers=None):
+def resolve_refs(spec, store=None, base_uri=""):
     """
     Resolve JSON references like {"$ref": <some URI>} in a spec.
     Optionally takes a store, which is a mapping from reference URLs to a
@@ -77,8 +78,7 @@ def resolve_refs(spec, store=None, handlers=None):
     """
     spec = deepcopy(spec)
     store = store or {}
-    handlers = handlers or default_handlers
-    resolver = RefResolver("", spec, store, handlers=handlers)
+    resolver = RefResolver(base_uri, spec, store, handlers=handlers)
 
     def _do_resolve(node):
         if isinstance(node, Mapping) and "$ref" in node:
@@ -94,7 +94,7 @@ def resolve_refs(spec, store=None, handlers=None):
             except KeyError:
                 # resolve external references
                 with resolver.resolving(node["$ref"]) as resolved:
-                    return resolved
+                    return _do_resolve(resolved)
         elif isinstance(node, Mapping):
             for k, v in node.items():
                 node[k] = _do_resolve(v)

--- a/connexion/spec.py
+++ b/connexion/spec.py
@@ -5,6 +5,7 @@ This module defines Python interfaces for OpenAPI specifications.
 import abc
 import copy
 import json
+import os
 import pathlib
 import pkgutil
 from collections.abc import Mapping
@@ -71,11 +72,11 @@ def canonical_base_path(base_path):
 
 
 class Specification(Mapping):
-    def __init__(self, raw_spec):
+    def __init__(self, raw_spec, *, base_uri=""):
         self._raw_spec = copy.deepcopy(raw_spec)
         self._set_defaults(raw_spec)
         self._validate_spec(raw_spec)
-        self._spec = resolve_refs(raw_spec)
+        self._spec = resolve_refs(raw_spec, base_uri=base_uri)
 
     @classmethod
     @abc.abstractmethod
@@ -145,13 +146,13 @@ class Specification(Mapping):
             return yaml.safe_load(openapi_string)
 
     @classmethod
-    def from_file(cls, spec, arguments=None):
+    def from_file(cls, spec, *, arguments=None, base_uri=""):
         """
         Takes in a path to a YAML file, and returns a Specification
         """
         specification_path = pathlib.Path(spec)
         spec = cls._load_spec_from_file(arguments, specification_path)
-        return cls.from_dict(spec)
+        return cls.from_dict(spec, base_uri=base_uri)
 
     @staticmethod
     def _get_spec_version(spec):
@@ -173,7 +174,7 @@ class Specification(Mapping):
         return version_tuple
 
     @classmethod
-    def from_dict(cls, spec):
+    def from_dict(cls, spec, *, base_uri=""):
         """
         Takes in a dictionary, and returns a Specification
         """
@@ -187,16 +188,17 @@ class Specification(Mapping):
         spec = enforce_string_keys(spec)
         version = cls._get_spec_version(spec)
         if version < (3, 0, 0):
-            return Swagger2Specification(spec)
-        return OpenAPISpecification(spec)
+            return Swagger2Specification(spec, base_uri=base_uri)
+        return OpenAPISpecification(spec, base_uri=base_uri)
 
     def clone(self):
-        return type(self)(copy.deepcopy(self._raw_spec))
+        return type(self)(copy.deepcopy(self._spec))
 
     @classmethod
-    def load(cls, spec, arguments=None):
+    def load(cls, spec, *, arguments=None):
         if not isinstance(spec, dict):
-            return cls.from_file(spec, arguments=arguments)
+            base_uri = f"{pathlib.Path(spec).parent}{os.sep}"
+            return cls.from_file(spec, arguments=arguments, base_uri=base_uri)
         return cls.from_dict(spec)
 
     def with_base_path(self, base_path):

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -48,6 +48,11 @@ def json_datetime_dir():
     return FIXTURES_FOLDER / "datetime_support"
 
 
+@pytest.fixture(scope="session")
+def relative_refs():
+    return FIXTURES_FOLDER / "relative_refs"
+
+
 @pytest.fixture(scope="session", params=SPECS)
 def spec(request):
     return request.param

--- a/tests/fixtures/relative_refs/components.yaml
+++ b/tests/fixtures/relative_refs/components.yaml
@@ -1,0 +1,37 @@
+components:
+  schemas:
+    Pet:
+      required:
+        - name
+      properties:
+        name:
+          type: string
+          example: fluffy
+        tag:
+          type: string
+          example: red
+        id:
+          type: integer
+          format: int64
+          readOnly: true
+          example: 1
+        last_updated:
+          type: string
+          readOnly: true
+          example: 2019-01-16T23:52:54.309102Z
+
+    Pets:
+      type: array
+      items:
+        $ref: "#/components/schemas/Pet"
+
+    Error:
+      properties:
+        code:
+          type: integer
+          format: int32
+        message:
+          type: string
+      required:
+        - code
+        - message

--- a/tests/fixtures/relative_refs/definitions.yaml
+++ b/tests/fixtures/relative_refs/definitions.yaml
@@ -1,0 +1,29 @@
+definitions:
+  Pet:
+    type: object
+    properties:
+      id:
+        type: integer
+        format: int64
+      name:
+        type: string
+      registered:
+        type: string
+        format: date-time
+
+  Pets:
+    type: array
+    items:
+      $ref: "#/definitions/Pet"
+
+  Error:
+    type: object
+    properties:
+      code:
+        type: integer
+        format: int32
+      message:
+        type: string
+    required:
+      - code
+      - message

--- a/tests/fixtures/relative_refs/openapi.yaml
+++ b/tests/fixtures/relative_refs/openapi.yaml
@@ -1,0 +1,45 @@
+openapi: 3.0.0
+
+info:
+  version: 1.0.0
+  title: Swagger Petstore
+  license:
+    name: MIT
+
+servers:
+  - url: /openapi
+
+paths:
+  /pets:
+    get:
+      summary: List all pets
+      responses:
+        '200':
+          description: A paged array of pets
+          content:
+            application/json:
+              schema:
+                $ref: "components.yaml#/components/schemas/Pets"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "components.yaml#/components/schemas/Error"
+
+  '/pets/{petId}':
+    get:
+      summary: Info for a specific pet
+      responses:
+        '200':
+          description: Expected response to a valid request
+          content:
+            application/json:
+              schema:
+                $ref: "components.yaml#/components/schemas/Pet"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "components.yaml#/components/schemas/Error"

--- a/tests/fixtures/relative_refs/swagger.yaml
+++ b/tests/fixtures/relative_refs/swagger.yaml
@@ -1,0 +1,36 @@
+swagger: "2.0"
+
+info:
+  title: "{{title}}"
+  version: "1.0"
+
+basePath: /swagger
+
+paths:
+  /pets:
+    get:
+      summary: List all pets
+      responses:
+        '200':
+          description: A paged array of pets
+          schema:
+            type: array
+            items:
+              $ref: 'definitions.yaml#/definitions/Pets'
+        default:
+          description: Unexpected Error
+          schema:
+            $ref:  'definitions.yaml#/definitions/Error'
+
+  '/pets/{id}':
+    get:
+      summary: Info for a specific pet
+      responses:
+        '200':
+          description: Expected response to a valid request
+          schema:
+            $ref: 'definitions.yaml#/definitions/Pet'
+        default:
+          description: Unexpected Error
+          schema:
+            $ref:  'definitions.yaml#/definitions/Error'

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -6,7 +6,7 @@ from unittest.mock import MagicMock
 import pytest
 from connexion import FlaskApi
 from connexion.exceptions import InvalidSpecification
-from connexion.spec import canonical_base_path
+from connexion.spec import Specification, canonical_base_path
 from yaml import YAMLError
 
 TEST_FOLDER = pathlib.Path(__file__).parent
@@ -136,6 +136,12 @@ def test_validation_error_on_completely_invalid_swagger_spec():
     with pytest.raises(InvalidSpecification):
         FlaskApi(pathlib.Path(f.name), base_path="/api/v1.0")
     os.unlink(f.name)
+
+
+def test_relative_refs(relative_refs, spec):
+    spec_path = relative_refs / spec
+    specification = Specification.load(spec_path)
+    assert "$ref" not in specification.raw
 
 
 @pytest.fixture

